### ISSUE TITLE
"if case" statements with multiple conditions are not handled properly

### ISSUE THREAD

**Issue by lc-bs** [nicklockwood/SwiftFormat#906]
With any `if case ...` statement with multiple conditions like the following one:
```swift
if let oldCollectionStatus = oldModel?.collectionStatus,
   case .updating(let oldTracks) = oldCollectionStatus,
   case .updated(let currentTracks) = model.collectionStatus,
   oldTracks.count == currentTracks.count,
   currentTracks.count < model.collection.tracksCount {
  ...
}
```
SwiftFormat treats it like a sort of switch, mistakenly changing the above example to:
```swift
if let oldCollectionStatus = oldModel?.collectionStatus,
   case .updating(let oldTracks) = oldCollectionStatus,
   case .updated(let currentTracks) = model.collectionStatus
case oldTracks.count == currentTracks.count
case currentTracks.count < model.collection.tracksCount {
  ...
}
```
(including the removal of commas and change of indentation, just like shown), which is of course wrong and doesn't compile.
The only workaround I found to this was to put the conditions without the `case` in a nested if statement.

**Comment by nicklockwood** [comment]
@lc-bs I wasn't able to reproduce this with just the code sample you provided. Can you provide a complete sample that reproduces the issue when formatted in isolation?

**Comment by lc-bs** [comment]
@nicklockwood sure, here's a simpler example that is still enough to reproduce the bug:
```swift
enum TestEnumWithValues {
  case firstCase
  case secondCase(value: [Int])
}

var firstEnumVar = TestEnumWithValues.firstCase
var secondEnumVar = TestEnumWithValues.secondCase(value: [1, 2, 3])

func testFunc() {
  if case .firstCase = firstEnumVar,
     case .secondCase(let value) = secondEnumVar,
     value.isEmpty {
    print("if body")
  }
}
```

For reference, here's the rules we're using (in .txt so that GitHub accepts the file):
[swiftformat.txt](https://github.com/nicklockwood/SwiftFormat/files/6342357/swiftformat.txt)

**Comment by nicklockwood** [comment]
@lc-bs awesome, thanks - I'm able to reproduce it now. FWIW, you can solve the problem temporarily by disabling the `wrapEnumCases` rule. You can either do that globally in your .swiftformat file, or just in that file, with a comment:

```swift
// swiftformat:disable wrapEnumCases
```

**Comment by nicklockwood** [comment]
@lc-bs fixed in 0.48.0
